rust/arrow/src/ipc/reader.rs

```diff
@@ -663,6 +664,21 @@ impl<R: Read> RecordBatchReader for StreamReader<R> {
     }
 }
 
+pub fn schema_from_bytes(bytes: &[u8]) -> Option<Schema> {
+    let ipc = ipc::get_root_as_message(&bytes[..]);
+    ipc.header_as_schema().map(|schema| fb_to_schema(schema))
+}
+
+pub fn recordbatch_from_bytes(
```

Comment:
you'd need the `data_header` and `data_body` to read a record batch correctly. See my commit fixing this.

Reply:
well that looks much better! Thanks for doing that.